src/main/java/com/aws/greengrass/logmanager/model/LogFileV2.java

```diff
@@ -0,0 +1,203 @@
+package com.aws.greengrass.logmanager.model;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Optional;
+
+
+/**
+ * Represents a log file in the file system via a hardlink to said file.
+ */
+public final class LogFileV2 {
+    public final Path hardLink;
+    private Path sourcePath;
+    private static final String HASH_VALUE_OF_EMPTY_STRING = "";
+    private static final int bytesNeeded = 1024;
+
+    private static final Logger logger = LogManager.getLogger(LogFileV2.class);
+    private String hash;
+
+
+    private LogFileV2(Path hardLink, Path sourcePath) {
+       this.hardLink = hardLink;
+       this.sourcePath = sourcePath;
+       this.hash = HASH_VALUE_OF_EMPTY_STRING;
+    }
+
+    /**
+     * Creates a LogFile from a regular java File. LogFile instances create a hardlink to a file and track it
+     * regardless of the original file being rotated. One important consideration is that getting the original source
+     * path if the original file rotates is only possible if the file remains on the same folder it originally was
+     * created. Rotation mechanisms that rotate files to different folders will yield in an invalid source path
+     * and a IOException will be thrown.
+     *
+     * @param sourceFile - the source java File.
+     * @param hardLink - a File path to where the file hardlink should be stored
+     * @throws IOException - if we fail to get the source file path
+     */
+    public static LogFileV2 fromFile(File sourceFile, Path hardLink) throws IOException {
```

Comment:
Do we need this? I think we can reuse the original `LogFile` -- it doesn't need to change at all, actually. In fact we can revert the SeekableByteChannel change from the weekend and just open the file directly.

LogFileGroup will make sure LogFiles are hard links instead of pointing directly at the target

Thoughts?

Reply:
I think we both have the same thoughts my idea was first write something and prove it could work with tests and then replace it. We would reuse the original log file but I just wanted to first create the constructs and then replace it in places little by little